test/forms/helpers/read-files-as-data-urls.test.js

```diff
@@ -0,0 +1,35 @@
+import { readFilesAsDataUrls } from '../../../src'
+
+describe('readFilesAsDataUrls', () => {
+  test('returns an array of file objects', async () => {
+    const file = new File(['content'], 'fileName.png', { type: 'image/png' })
+    const result = await readFilesAsDataUrls([file])
+    expect(Object.keys(result.at(0))).toEqual(
+      expect.arrayContaining(['name', 'size', 'type', 'lastModified', 'url'])
+    )
+    expect(Object.values(result.at(0)).every(Boolean)).toBe(true)
+  })
+
+  test('returns an empty url when run on the server', async () => {
+    const file = new File(['content'], 'fileName.png', { type: 'image/png' })
+
+    const originalWindow = setWindowValue(undefined)
+    const result = await readFilesAsDataUrls([file])
+    expect(result.at(0).url).toEqual('')
+
+    setWindowValue(originalWindow)
+  })
+})
+
+// HELPERS
+
+function setWindowValue(value) {
+  // eslint-disable-next-line no-undef
+  const originalWindow = global.window
+  // eslint-disable-next-line no-undef
+  Object.defineProperty(global, 'window', {
+    value
+  })
```

Comment:
Setting `global.window = undefined` directly doesn't work, since I believe it's considered read only. This is a hacky way around it. LMK if you know any other "official" approaches.

I went down the rabbit hole of trying to mock a function within a function (i.e., `isServer`), but there doesn't seem to be a clear way to do that in Jest (without also exporting that local function from the module). 

https://stackoverflow.com/questions/39755439/how-to-mock-imported-named-function-in-jest-when-module-is-unmocked

Reply:
We do have a handful of components with this guard in place, so it might make sense to spend time in the future investigating how to do this "properly".

Reply:
Nevermind, I think I figured out a better solution!